elastalert/test_rule.py

```diff
@@ -222,10 +217,9 @@ def get_id():
             endtime = ts_now()
             starttime = endtime - datetime.timedelta(days=args.days)
 
-        # Set run_every to cover the entire time range unless count query, terms query or agg query used
-        # This is to prevent query segmenting which unnecessarily slows down tests
-        if not rule.get('use_terms_query') and not rule.get('use_count_query') and not rule.get('aggregation_query_element'):
-            conf['run_every'] = endtime - starttime
+        # Set run_every to cover the entire time range, this is to prevent query segmenting which unnecessarily slows down tests
+        # (see elastalert.get_segment_size())
+        conf['run_every'] = endtime - starttime
```

Comment:
I'm a little bit conflicted here, because seeing 
```
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:14 PDT to 2017-03-29 16:15 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:15 PDT to 2017-03-29 16:16 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:16 PDT to 2017-03-29 16:17 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:17 PDT to 2017-03-29 16:18 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:18 PDT to 2017-03-29 16:19 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:19 PDT to 2017-03-29 16:20 PDT: 0 hits
INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:20 PDT to 2017-03-29 16:21 PDT: 0 hits
```
can be really slow and confusing to some people, it more mimics the real behavior. If someone has a rule with a small timeframe, this accuracy is crucial.

Seeing ``INFO:elastalert:Queried rule Example Rule from 2017-03-29 16:01 PDT to 2017-03-30 16:01 PDT: 0 hits`` is super quick, and useful to see the total number of matching documents, but won't trigger alerts in the same way.

The latter is almost equivalent to running it with --count-only. For non count/terms queries,  segment size doesn't matter, because each document is processed individually. That optimization was added to just make the output look nicer.

If we can't set run_every per rule, I'd rather see it not changed at all.